packages/tools/devtools/devtools-browser-extension/src/background/BackgroundScript.ts

```diff
@@ -64,6 +65,15 @@ browser.runtime.onConnect.addListener((devtoolsPort: Port): void => {
 			);
 			return;
 		}
+		if (message.type === GetDevtoolsFeatures.MessageType) {
+			const port = chrome.runtime.connect({ name: "popup-background-communication" });
+			relayMessageToPort(
```

Comment:
Instead of using this message, I think the relay should go to the active tab. Maybe by using `browser.tabs.connect`?

Reply:
I believe you should be able to get the tab ID from within the popup script, since it (like the devtools panel) is associated with a single tab instance (I _think_). In which case, we probably want to define a new message specifically for the popup script to sent to the background script (e.g. DEVTOOLS_EXISTANCE_CHECK), which will include the tab ID. The background script can then have a completely separate listener for these messages (rather than running in `browser.runtime.onConnect`, which is only there for connection from the devtools script). Happy to chat about this in more detail if you'd like.